### ISSUE TITLE
Handle onCancelled in FirebaseArray

### ISSUE THREAD

**Issue by davideast** [firebase/FirebaseUI-Android#71]
[We currently ignore](log) any `onCancelled` errors inside of FirebaseArray. At the very least we should log them to LogCat.

```java
    public void onCancelled(FirebaseError firebaseError) {
        Log.v(LOG_TAG, firebaseError.getMessage());
    }
```

[log]: https://github.com/firebase/FirebaseUI-Android/blob/ea3cb58454756ef417ad4f23307bc7143d19fdfb/library/src/main/java/com/firebase/ui/FirebaseArray.java#L111-L113

**Comment by puf** [comment]
As far as I know, the only reason `onCancelled()` can be called is because
the user has no read permission on the location. If that is the case, then
it signals a programming error when this is called and just logging it is a
sure-fire way to get it ignored.

Did you try what happens when you throw the exception?

If we log it, we should at least log it as a warning, but possibly an error.

**Comment by abehaskins** [comment]
I'm in favor of allowing users to handle onCancelled through normal firebase auth flow and acting as if we have proper access, but I'm open to discussion.

**Comment by puf** [comment]
For another reason, I was yesterday hacking on a "secured chat" scenario
that triggers this error:
https://github.com/firebase/FirebaseUI-Android/compare/puf-secured-chat

I have no opinion yet on how I want it to work there, but the branch might
be useful to play with.

**Comment by SUPERCILEX** [comment]
With the new Firebase crash reporting, I think it would be easiest to just log the error: `FirebaseCrash.report(databaseError.toException());`

**Comment by samtstern** [comment]
@SUPERCILEX that's a good option, but it probably is not worth giving this library a dependency on `FirebaseCrash` to report that sort of exception.

**Comment by puf** [comment]
We probably should surface onCancelled() in some way, so that your app
could report the error if it wanted.

**Comment by SUPERCILEX** [comment]
@samtstern Oh, my bad. For some reason, I thought we were already including `FirebaseCrash`.

@puf The big issue I see with my PR is that is breaks existing code. If you're using android studio, it's not that bad because they have a quick fix to implement new methods, but still. Can you guys think of a better way to do this? If not, I think it's better to do this before 1.0.0.

**Comment by samtstern** [comment]
@SUPERCILEX there's no problem with the breaking change, we'll just have to do a new version for this.  Likely `0.6.0` in following our tradition of minor-version breaking changes in this pre-1.0 state, but I will have to ask around the team to confirm that.

**Comment by puf** [comment]
With my proposal of a default implementation, this will no longer be a
breaking change. Right?

Or are we concerned about another part of the change?

**Comment by samtstern** [comment]
This has been released in version `0.5.2`.
